test/openshift-ci/setup-e2e-tests.sh

```diff
@@ -0,0 +1,208 @@
+  ./$KSERVE_DIR/test/scripts/openshift-ci/deploy.serverless.sh
+fi
+
+echo "Installing KServe with Minio"
+kustomize build $KSERVE_DIR/config/overlays/test |
+  sed "s|kserve/storage-initializer:latest|${STORAGE_INITIALIZER_IMAGE}|" |
+  sed "s|kserve/agent:latest|${KSERVE_AGENT_IMAGE}|" |
+  sed "s|kserve/router:latest|${KSERVE_ROUTER_IMAGE}|" |
+  sed "s|kserve/kserve-controller:latest|${KSERVE_CONTROLLER_IMAGE}|" |
+  oc apply --server-side=true -f -
+
+# Install DSC/DSCI for test. (sometimes there is timing issue when it is under the same kustomization so it is separated)
+oc create -f $KSERVE_DIR/config/overlays/test/dsci.yaml
+oc create -f $KSERVE_DIR/config/overlays/test/dsc.yaml
+
+# Patch the inferenceservice-config ConfigMap, when running RawDeployment tests
+if [ "$1" == "raw" ]; then
+  export OPENSHIFT_INGRESS_DOMAIN=$(oc get ingresses.config cluster -o jsonpath='{.spec.domain}')
+  oc patch configmap inferenceservice-config -n kserve --patch-file <(cat $KSERVE_DIR/config/overlays/test/configmap/inferenceservice-openshift-ci-raw.yaml | envsubst)
+  oc delete pod -n kserve -l control-plane=kserve-controller-manager
+
+  oc patch DataScienceCluster test-dsc --type='json' -p='[{"op": "replace", "path": "/spec/components/kserve/defaultDeploymentMode", "value": "RawDeployment"}]'
+else
+  export OPENSHIFT_INGRESS_DOMAIN=$(oc get ingresses.config cluster -o jsonpath='{.spec.domain}')
+  oc patch configmap inferenceservice-config -n kserve --patch-file <(cat $KSERVE_DIR/config/overlays/test/configmap/inferenceservice-openshift-ci-serverless.yaml | envsubst)
+fi
+
+
+# Determine the odh model-controller image from params.env
+# so it can be used in the sed command later to replace with the PR image
+source $ODH_MODEL_CTRL_DIR/config/base/params.env
+image_to_replace=$odh-model-controller
+
+echo "Installing odh-model-controller dependencies"
+# authorino
+curl -sL https://raw.githubusercontent.com/Kuadrant/authorino-operator/main/utils/install.sh | sed "s|kubectl|oc|" |
+  bash -s -- -v 0.16.0
+
+# kserve-local-gateway
+curl https://raw.githubusercontent.com/opendatahub-io/opendatahub-operator/bde4b4e8478b5d03195e2777b9d550922e3cdcbc/components/kserve/resources/servicemesh/routing/istio-kserve-local-gateway.tmpl.yaml |
+    sed "s/{{ .ControlPlane.Namespace }}/istio-system/g" |
+    oc create -f -
+
+curl https://raw.githubusercontent.com/opendatahub-io/opendatahub-operator/bde4b4e8478b5d03195e2777b9d550922e3cdcbc/components/kserve/resources/servicemesh/routing/kserve-local-gateway-svc.tmpl.yaml |
+    sed "s/{{ .ControlPlane.Namespace }}/istio-system/g" |
+    oc create -f -
+
+kustomize build $ODH_MODEL_CTRL_DIR/config/base |
+  sed "s|${image_to_replace}|${ODH_MODEL_CONTROLLER_IMAGE}|" |
+  oc apply --server-side=true -n kserve -f -
+
+oc wait --for=condition=ready pod -l app=odh-model-controller -n kserve --timeout=300s
+
+echo "Add testing models to minio storage ..." # Reference: config/overlays/test/minio/minio-init-job.yaml
```

Comment:
consider using our minio image that we have on quay.

Reply:
I would prefer to keep the test setups as similar as possible between the kserve repo and this repo, wdyt?